projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-sorting.component.ts

```diff
@@ -31,8 +31,16 @@ export class IgxExcelStyleSortingComponent {
     constructor() {}
 
     public onSortButtonClicked(sortDirection) {
+        let isColumnGrouped;
+        if (this.grid instanceof IgxGridComponent) {
```

Comment:
Excel style filtering is applicable to all grids we currently have. Casting the current grid instance to one grid type is not a good solution.
In my opinion you should add a method in the base class which will return if a given column from the current grid instance is grouped or not. This method will return actual result only for the grid and for the hierarchical and tree-grid it will return false.
That way we wont make a dependency to one grid type in the feature which is applicable to all grid types.

Reply:
The reason for this is that only IgxGrid supports grouping and only it has 'groupingExpressions' property.

Reply:
Sure, but my comment is still valid. We shouldn't introduce dependency to one grid type here. Tomorrow some of the other grids may support grouping and this code will fail.

Reply:
Ok, I got your point, I will add a getter in the base class that will return false and I will override it in the IgxGrid only.